daemon/container.go

```diff
@@ -1090,29 +1093,109 @@ func (container *Container) mountVolumes() error {
 	return nil
 }
 
-func (container *Container) copyImagePathContent(v volume.Volume, destination string) error {
-	rootfs, err := symlink.FollowSymlinkInScope(filepath.Join(container.basefs, destination), container.basefs)
-	if err != nil {
-		return err
+func (container *Container) prepareMountPoints() error {
+	for _, config := range container.MountPoints {
+		if len(config.Driver) > 0 {
+			v, err := container.daemon.createVolume(config.Name, config.Driver, nil)
+			if err != nil {
+				return err
+			}
+			config.Volume = v
+		}
 	}
+	return nil
+}
 
-	if _, err = ioutil.ReadDir(rootfs); err != nil {
-		if os.IsNotExist(err) {
-			return nil
+func (container *Container) removeMountPoints(rm bool) error {
+	var rmErrors []string
+	for _, m := range container.MountPoints {
+		if m.Volume == nil {
+			continue
+		}
+		container.daemon.volumes.Decrement(m.Volume)
+		if rm {
+			err := container.daemon.volumes.Remove(m.Volume)
+			// ErrVolumeInUse is ignored because having this
+			// volume being referenced by othe container is
```

Comment:
nit: (sp) `other` -- No need to update just for this -- could be done if other changes are needed.